tools/optimizer/optimizer.cpp

```diff
@@ -716,7 +716,7 @@ Ref simplifyCondition(Ref node) {
 
 StringSet ELIMINATION_SAFE_NODES("assign call if toplevel do return label switch binary unary-prefix"); // do is checked carefully, however
 StringSet IGNORABLE_ELIMINATOR_SCAN_NODES("num toplevel string break continue dot"); // dot can only be STRING_TABLE.*
-StringSet ABORTING_ELIMINATOR_SCAN_NODES("new object function defun for while array throw"); // we could handle some of these, TODO, but nontrivial (e.g. for while, the condition is hit multiple times after the body)
+StringSet ABORTING_ELIMINATOR_SCAN_NODES("new object defun while array"); // we could handle some of these, TODO, but nontrivial (e.g. for while, the condition is hit multiple times after the body)
```

Comment:
actually, since this is just for the eliminator pass, it will only see the contents of an asm.js function. `new, object, defun` and `array` are all invalid in an asm.js function, so really only `while` is left...